the-collab-lab/tcl-6-smart-shopping-list
# Generate token and save in localStorage.

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |


## Description



A user without a token is shown a button, `Create New List`. Clicking this button generates and saves a token to `localStorage`. App state is updated and the `Shopping` component re-renders showing a shopping list and navigation buttons.

![Screen Shot 2020-04-14 at 1 33 01 PM](https://user-images.githubusercontent.com/7929227/79271429-a4797f80-7e54-11ea-89ea-f1bd0d022252.png)

![Screen Shot 2020-04-14 at 1 33 31 PM](https://user-images.githubusercontent.com/7929227/79271442-a93e3380-7e54-11ea-91be-5147cf02c079.png)

![Screen Shot 2020-04-14 at 1 33 51 PM](https://user-images.githubusercontent.com/7929227/79271457-b0654180-7e54-11ea-967e-e7f341033b0f.png)

## Related Issue


Closes #5 

## Acceptance Criteria

A shopping list consists of a set of items associated with a user’s token. Creating a new list consists of the following:

* Generate a new, unique token
* Save the token to `localStorage`
* Show the user the list view

The following script can be used to generate a suitable token: https://gist.github.com/segdeha/21a42618ce5a54916c5b58d36ec2992e

AC:

* For users who do not already have a token/list, a button or link exists at the `/` route (i.e. home screen) that allows them to create a new list
* Clicking the button/link generates a new token and saves it to `localStorage`
* Once the token has been created and saved, the user is redirected to the "list" view


## Testing Steps / QA Criteria



1. From your terminal, pull down the branch by using `git pull origin sb-sk-new-shopping-list` and `git checkout sb-sk-new-shopping-list` to switch to the branch.
2. Run `npm install` to add the dependencies, then `npm start` to run the app.
3. The app will run in the browser and is set to look for a token named `user_token` in `localStorage`. 
4. If there is **not** a token the view shows a `Create New List` button. When clicked a token is generated and saved to `localStorage`, and the view re-renders to show a shopping list.  
5. It's not likely that you already have a token with the name `user_token` the first time you run the app, but if you do or when you want to check out how the app is working more than once, run `localStorage.removeItem('user_token')` in your browser's console. You will need to refresh the page to get back to the start.
